components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -174,31 +177,36 @@ private void validateDateOfBirthClaimValue(String claimURI, String claimValue, U
 
         if (StringUtils.equalsIgnoreCase(DATE_OF_BIRTH_LOCAL_CLAIM, claimURI)) {
             String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantId());
-            String dateOfBirthRegex = getDateOfBirthRegex(tenantDomain);
+            Map<String, String> dateOfBirthClaimProperties = getDateOfBirthClaimProperties(tenantDomain);
```

Comment:
Can `dateOfBirthClaimProperties` be null?

Reply:
No, it returns an empty array https://github.com/wso2/carbon-identity-framework/blob/8545eb7a19fffe6f157c6657bf36b2be7e18e18a/components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/model/Claim.java#L69